[IMP] l10n_ar_account_tax_settlement, l10n_ar_tax_settlement_backward_comp: Importe otros conceptos en txt agip en pagos migrados. Impuesto en apunte contable

pablohmontenegro opened the pull request:
Commit 1) l10n_ar_account_tax_settlement: Buscamos los payments sin retención que vienen migrados de la versión anterior y le sumamos el amount total de los mismos (move_id.amount_total_in_currency_signed) al total_amount de la retención. Esto lo hacemos porque en la migración de 16 a 18 se migran los pagos y las retenciones por separado a diferencia de 16 que estaba todo en el mismo asiento.
Commit 2) l10n_ar_tax_settlement_backward_comp: Si el impuesto del apunte está activo entonces no fue migrado (el apunte fue creado en la nueva versión) por lo tanto devolvemos el impuesto del apunte y no lo buscamos en el contacto al momento de generar txt de impuestos.
Explicación de los cambios: https://app.screencastify.com/watch/QqTioopqRZExw8KcZNFv (comparto enlace de drive por las dudas https://drive.google.com/file/d/1QENDDYSkgZhcRc254YsCBEMlel-1ZzFn/view )
Ticket Adhoc side: 100241

pablohmontenegro committed:
[IMP] l10n_ar_account_tax_settlement: Importe otros conceptos en txt agip en pagos migrados.
Buscamos los payments sin retención que vienen migrados de la versión anterior y le sumamos el amount total de los mismos (move_id.amount_total_in_currency_signed) al total_amount de la retención. Esto lo hacemos porque en la migración de 16 a 18 se migran los pagos y las retenciones por separado a diferencia de 16 que estaba todo en el mismo asiento.
Ticket Adhoc side: 100241

roboadhoc commented:
[![Pull request status dashboard](https://mergebot.adhoc.ar/ingadhoc/odoo-argentina-ee/pull/703.png)](https://mergebot.adhoc.ar/ingadhoc/odoo-argentina-ee/pull/703)

pablohmontenegro commented on l10n_ar_account_tax_settlement/models/account_journal.py:
```diff
@@ -571,6 +571,23 @@ def iibb_aplicado_agip_files_values(self, move_lines):  # noqa: C901
+                    # Buscamos los payments que sin retención que vienen migrados de la versión anterior y le sumamos
+                    # el amount total de los mismos (move_id.amount_total_in_currency_signed) al total_amount de la
+                    # retención. Esto lo hacemos porque en la migración de 16 a 18 se migran los pagos y las retenciones
+                    # por separado a diferencia de 16 que estaba todo en el mismo asiento.
+                    related_payments = self.env["account.payment"].search(
+                        [
+                            ("name", "=", payment.name),
+                            ("company_id", "=", payment.company_id.id),
+                            ("partner_id", "=", payment.partner_id.id),
+                            ("id", "!=", payment.id),
+                        ]
+                    )
```
I think the only way to find payments without withholding is to search by name

pablohmontenegro commented on l10n_ar_account_tax_settlement/models/account_journal.py:
```diff
@@ -571,6 +571,23 @@ def iibb_aplicado_agip_files_values(self, move_lines):  # noqa: C901
+                            ("name", "=", payment.name),
+                            ("company_id", "=", payment.company_id.id),
+                            ("partner_id", "=", payment.partner_id.id),
+                            ("id", "!=", payment.id),
```
I added only ("state", "=", "posted") because I guess ("date", "=", payment.date) may not find the corresponding records if the payments have different dates.

pablohmontenegro committed:
[IMP] l10n_ar_tax_settlement_backward_comp: impuesto en apunte contable
Si el impuesto del apunte está activo entonces no fue migrado (el apunte fue creado en la nueva versión) por lo tanto devolvemos el impuesto del apunte y no lo buscamos en el contacto al momento de generar txt de impuestos.
Ticket Adhoc side: 100241

jue-adhoc commented:
@roboadhoc r+

roboadhoc commented:
@pablohmontenegro @jue-adhoc because this PR has multiple commits, I need to know how to merge it:

* `merge` to merge directly, using the PR as merge commit message
* `rebase-merge` to rebase and merge, using the PR as merge commit message
* `rebase-ff` to rebase and fast-forward

pablohmontenegro commented:
Cierro este pr porque da error en "ci/runbot-modified-modules" (https://runbot.dev-adhoc.com/runbot/build/13520) y no se por qué. Creé estos 2 prs que lo reemplazan https://github.com/ingadhoc/odoo-argentina-ee/pull/713 y https://github.com/ingadhoc/odoo-argentina-ee/pull/714
